### ISSUE TITLE
localhost:8080 will be parsed as localhost

### ISSUE THREAD

**Issue by alucard001** [apereo/phpCAS#238]
Here is how I test with Vagrant:

- Vagrant IP: 10.0.2.1, 
- HTTP_HOST: localhost
- `$_SERVER['SERVER_NAME']`: localhost
- Vagrant server port: 80

But since I am using Vagrant(Win), which means I use browser to browse: http://localhost:8080 in order to use phpCAS.

Now the problem is: I have a redirect URL (URL to go to after login) that will go to: 

`http://localhost:8080/some_url`

 (Browser URL).

Here is the problem: in `phpCAS/CAS/Client.php:3588`, it will check the emptiness of `$_SERVER['SERVER_NAME']` .  Since inside the vagrant box, the `$_SERVER['SERVER_NAME']` is localhost, which is not empty, result in the full URL become:

`http://loaclhost/some_url`

In other words, the port number is removed, causing CAS Auth fail because the requested redirect URL is different from the real redirect URL.

Only when I change from:

`if (empty($_SERVER['SERVER_NAME'])) `

to 

`if ($_SERVER['SERVER_NAME'] == 'localhost' || empty($_SERVER['SERVER_NAME'])) `

It will work.

I know this is not a correct way to do it.  But I hope that there is a better way to do it.

Thank you very much.

**Comment by jfritschi** [comment]
Please have a look at the detailed code docs:
https://developer.jasig.org/cas-clients/php/current/docs/api/classphpCAS.html

Check out the set* functions. With them you can override any automatically detected  and configured URL. The setFixedServiceURL ($url) is probably what you are looking for.

**Comment by alucard001** [comment]
Yes it works!

Thank you very much.
